catroid/src/main/java/org/catrobat/catroid/content/Sprite.java

```diff
@@ -249,10 +250,10 @@ public void createAndAddActions(@CreateActionsMode int includeStartActions) {
 	}
 
 	private void createAndAddAction(Script script, @CreateActionsMode int includeStartActions) {
-		if (script.isCommentedOut()) {
-			return;
-		} else if (!isClone && script instanceof StartScript
-				|| isClone && script instanceof WhenClonedScript) {
+		if (!script.isCommentedOut() && !isClone
+				&& script instanceof StartScript
+				|| isClone
+				&& script instanceof WhenClonedScript) {
```

Comment:
Since`&&`>`||`, braces are missing `!script.isCommentedOut() && (a || b)`. btw, the formatting is a little off here :wink:

However, I think this is a behavior change:
Previously, if the script is commented out, the method exits;
Now, if the script is commented out, the if block evaluates to false, however the else if block (`else if (script instanceof EventScript)`) will still be executed (but shouldn't).

I personally love guard clauses, but ___imo___ (note that the imo is underlined and italic! :wink:) it should be
```java
if (script.isCommentedOut()) {
	return;
}

if (!isClone && script instanceof…
```

If the script is commented out, early exit and we're done here.
If not, do the "real" logic.

Reply:
sure :wink